test/unit/aggregate_store_test.rb

```diff
@@ -85,13 +85,78 @@ def elasticsearch_store
         assert !passport.changed?
       end
 
-      should "respond to changed? appropriatetly when instance is not a active record object" do
+      should "respond to changed? appropriately when instance is not a active record object" do
         assert !@instance.changed?
 
         @instance.name = "blah"
         assert @instance.changed?
       end
 
+      Rails::VERSION::MAJOR > 4 && ['save', 'save!'].each do |save_method|
+        should "respond to saved_changes? appropriately when the instance is a active record object using #{save_method}" do
+          # change triggered by a active record attribute change
+
+          passport = sample_passport
+          passport.public_send(save_method)
+          refute passport.saved_changes?
+          passport.name = "blah"
+          refute passport.saved_changes?
+          passport.public_send(save_method)
+          assert passport.saved_changes?
+
+          passport = Passport.find(passport.id)
+
+          # change triggered by a aggregate attribute change
+          passport.foreign_visits = [ForeignVisit.new(country: "Canada"), ForeignVisit.new(country: "Mexico")]
+          passport.public_send(save_method)
+          assert passport.saved_changes?
+
+          passport = Passport.find(passport.id)
+
+          # not mark saved_changes if the aggregate value being assigned is same as before
+          visits = [ForeignVisit.new(country: "Spain"), ForeignVisit.new(country: "Japan")]
+          passport.foreign_visits = visits
+          passport.public_send(save_method)
+          assert passport.saved_changes?
+          passport.foreign_visits = visits
+          passport.public_send(save_method)
+          refute passport.saved_changes?
+        end
+
+        should "respond to saved_changes? appropriately when instance is not a active record object using #{save_method}" do
+          refute @instance.saved_changes?
+          @instance.name = "blah"
+
+          assert_raise(NoMethodError) { @instance.public_send(save_method) }
+          refute @instance.saved_changes?
+        end
+      end
+
+      Rails::VERSION::MAJOR === 4 && ['save', 'save!'].each do |save_method|
```

Comment:
Same comment about using the Active Record helpers.

Also, you're using `===` here, Ruby we just prefer `==`